### PR TITLE
Working references

### LINKED ISSUE #3: Connect instance script with template expressions for code insight
It will fix unused imports problem and improve rename refactoring

### PR DESCRIPTION
**Pull request by tomblachut**
This ~is a very rough take to get rid of~ removes faulty code injector and make whole plugin robust at last by using lazy parsing. No more weird inspections, switch to JSX prompts. Even intentions like `Remove parentheses` work.

Completion popups need to get care after this.

Resolves #3

### CODE REVIEW

**Comment by tomblachut** [comment]
Work on this PR gave me better outlook on the next steps:

* Attributes should also be handled by lazy parsing (Angular does this) - that requires tinkering with `SvelteHTMLLanguage` and `SvelteJSReferenceExpressionResolver` introduced in said #50. (#2)
* `<Foo {name}>` will require changes to HTML lexer/parser and also fixing missing required attributes inspections - VueXmlExtension#isRequiredAttributeImplicitlyPresent` i think handles that.
* `<Foo prop={name}>` without quotes is a similar story, I don't know yet but there can be issues with escape characters. (#22)
* For `use:function` to work `function` must have a reference. IDK about best approach here but for sure injection is not required. (#11)
* Work on `$` labels is blocked by me figuring out how to parse JS inside `script` as custom SvelteJS with working references. Currently I've switched JS from default dialect to ES6 module which fixes some things but isn't extensible. (#7)
* Store `$references` are tricky. I think conditional lazy parsing is the best course of action here - there is some prior art with Angular parsing `#ref` as `#`  & `ref`. I tried with reference contributor but find usages don't work then and extending those word scanners feels hacky. (#4)

File: src/main/java/dev/blachut/svelte/lang/SvelteJSReferenceExpressionResolver.kt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import com.intellij.psi.xml.XmlTag
+import com.intellij.xml.util.HtmlUtil
+import dev.blachut.svelte.lang.psi.*
+
+class SvelteJSReferenceExpressionResolver(referenceExpression: JSReferenceExpressionImpl,
+                                          ignorePerformanceLimits: Boolean) :
+    JSReferenceExpressionResolver(referenceExpression, ignorePerformanceLimits) {
+    override fun resolve(expression: JSReferenceExpressionImpl, incompleteCode: Boolean): Array<ResolveResult> {
+        return if (myUnqualifiedOrLocalResolve) {
+            resolveInComponent(expression, incompleteCode) ?: arrayOf()
```
Comment (unlocomqx) [inline src/main/java/dev/blachut/svelte/lang/SvelteJSReferenceExpressionResolver.kt]: Could fallback to `super.resolve(expression, incompleteCode)` to resolve expressions like `Math.pow(1, 2)`

Comment (tomblachut) [reply src/main/java/dev/blachut/svelte/lang/SvelteJSReferenceExpressionResolver.kt]: I'll probably do it like you've said for now. I'm tinkering with something better so there won't be false-positives but with no luck 😢 And better to release at least something something

Comment (unlocomqx) [reply src/main/java/dev/blachut/svelte/lang/SvelteJSReferenceExpressionResolver.kt]: Well, this looks awesome already! Anyway, I can help with debugging if something is acting really weird.

Comment (tomblachut) [reply src/main/java/dev/blachut/svelte/lang/SvelteJSReferenceExpressionResolver.kt]: I'm not taking about bugs. IDE sometimes picks up references from other files and links them. Now I've removed if from resolve and applied your suggestion and it magically started working better so I don't understand this stuff anymore 😅

**Pull request merged**